server/src/sas/CompletionProvider.ts

```diff
@@ -606,6 +628,39 @@ export class CompletionProvider {
               : `'${item.name.replace(/'/g, "''")}'n`,
         }));
 
+        if (
+          prefix.length > 1 &&
+          (items === undefined ||
+            items.length === 0 ||
+            !items.find((item) => item.label.startsWith(prefix))) &&
```

Comment:
It might be worth pulling out this condition into a variable that indicates what we're checking here